build_tools/python/e2e_test_artifacts/cmake_generator/iree_rule_generator.py

```diff
@@ -188,30 +189,53 @@ def _build_target_path(self, target_name: str):
 
 def generate_rules(
     package_name: str, root_path: pathlib.PurePath,
-    artifacts_root: iree_artifacts.ArtifactsRoot,
+    module_generation_configs: Sequence[
+        iree_definitions.ModuleGenerationConfig],
     model_rule_map: Dict[str, model_rule_generator.ModelRule]) -> List[str]:
-  """Generates all rules to build IREE artifacts."""
+  """Generates all rules to build IREE artifacts.
+  
+  Args:
+    package_name: CMake package name for rules.
+    root_path: path of the root artifact directory.
+    module_generation_configs: list of IREE module generation configs.
+    model_rule_map: map of generated model rules keyed by model id, it must
+      cover all model referenced in module_generation_configs.
+  Returns:
+    List of cmake rules.
+  """
 
   rule_builder = IreeRuleBuilder(package_name=package_name)
 
-  cmake_rules = []
-  for model_dir in artifacts_root.model_dir_map.values():
-    imported_model_artifact = model_dir.imported_model_artifact
-    imported_model = imported_model_artifact.imported_model
-    model_rule = model_rule_map[imported_model.model.id]
+  all_imported_models = collections.OrderedDict(
```

Comment:
As of Python 3.7, regular dictionaries are guaranteed to be ordered, so I don't think you need to use OrderedDict any longer

Reply:
I was intentionally using OrderedDict to provide a clearer semantic. I'm fine to use dict.
Let's keep it as for consistency and I can have another refactor to remove all OrderedDict usages.